proto/qdrant/cloud/payment/v1/payment.proto

```diff
@@ -0,0 +1,311 @@
+  StripeSession stripe_session = 1;
+}
+
+// Represents a Stripe Session abstraction containing Checkout session and SetupIntent references.
+// buf:lint:ignore QDRANT_CLOUD_REQUIRED_ENTITY_FIELDS
+message StripeSession {
+  // The unique identifier of the Stripe Checkout session.
+  string id = 1;
+  // The URL to redirect the user to complete the Checkout session.
+  // This may be null if the session is incomplete or improperly initialized.
+  optional string url = 2;
+  // The Stripe customer ID associated with this session.
+  string customer = 3;
+  // The ID of the associated Stripe SetupIntent.
+  string setup_intent_id = 4;
+  // The status of the associated Stripe SetupIntent.
+  SetupIntentStatus setup_intent_status = 5;
+  // The payment method attached to the associated SetupIntent.
+  string setup_intent_payment_method = 6;
+}
+
+// Represents the status of a SetupIntent in Stripe.
+enum SetupIntentStatus {
+  // Default unspecified value.
+  SETUP_INTENT_STATUS_UNSPECIFIED = 0;
+  // A payment method is required to proceed.
+  SETUP_INTENT_STATUS_REQUIRES_PAYMENT_METHOD = 1;
+  // The SetupIntent is ready to be confirmed.
+  SETUP_INTENT_STATUS_REQUIRES_CONFIRMATION = 2;
+  // The SetupIntent requires further action (e.g., 3D Secure).
+  SETUP_INTENT_STATUS_REQUIRES_ACTION = 3;
+  // The SetupIntent is being processed.
+  SETUP_INTENT_STATUS_PROCESSING = 4;
+  // The SetupIntent has been canceled.
+  SETUP_INTENT_STATUS_CANCELED = 5;
+  // The SetupIntent has succeeded.
+  SETUP_INTENT_STATUS_SUCCEEDED = 6;
+}
+
+// GetStripeSessionRequest is the request for the GetStripeSession function
+message GetStripeSessionRequest {
+  // The identifier of the account (in GUID format).
+  // This is a required field.
+  string account_id = 1 [(buf.validate.field).string = {uuid: true}];
+  // The identifier of the Stripe session.
+  // This is a required field.
+  string session_id = 2;
+}
+
+// GetStripeSessionResponse is the response from the GetStripeSession function
+message GetStripeSessionResponse {
+  // The Stripe session for the account.
+  StripeSession stripe_session = 1;
+}
+
+// ChangePaymentInformationRequest is the request for the ChangePaymentInformation function
+message ChangePaymentInformationRequest {
+  // The identifier of the account (in GUID format).
+  string account_id = 1 [(buf.validate.field).string = {uuid: true}];
+  // The identifier of the new payment information to set.
+  string new_payment_information_id = 3 [(buf.validate.field).string = {min_len: 1}];
+}
+
+// ChangePaymentInformationResponse is the response returned after a successful update.
+message ChangePaymentInformationResponse {
+  // New payment information that has been set for the account.
+  PaymentInformation payment_information = 1;
+}
+
+// Represents a payment information.
+// buf:lint:ignore QDRANT_CLOUD_REQUIRED_ENTITY_FIELDS
+// TODO: Is payment method a better name?
+message PaymentInformation {
```

Comment:
To me `PaymentInformation` sounds a bit too vague. In this case we're just using it as a wrapper for a payment method. Because of that, I'd lean toward naming it directly `PaymentMethod`.